### PR TITLE
Fix: can_bundle.py chooses an arbitrary default source (#2818)

### PR DESCRIPTION
**Pull request by achave11**
Author

- [x] PR title references issue
- [x] Title of main commit references issue
- [x] PR is linked to Zenhub issue

Author (reindex)

- [x] Added `r` tag to commit title                         <sub>or this PR does not require reindexing</sub>
- [x] Added `reindex` label to PR                           <sub>or this PR does not require reindexing</sub>

Author (freebies & chains)

- [x] Freebies are blocked on this PR                       <sub>or there are no freebies in this PR</sub>
- [x] Freebies are referenced in commit titles              <sub>or there are no freebies in this PR</sub>
- [x] This PR is blocked by previous PR in the chain        <sub>or this PR is not chained to another PR</sub>
- [x] Added `chain` label to the blocking PR                <sub>or this PR is not chained to another PR</sub>

Author (upgrading)

- [x] Documented upgrading of deployments in UPGRADING.rst  <sub>or this PR does not require upgrading</sub>
- [x] Added `u` tag to commit title                         <sub>or this PR does not require upgrading</sub>
- [x] Added `upgrade` label to PR                           <sub>or this PR does not require upgrading</sub>

Author (requirements, before every review)

- [x] Ran `make requirements_update`                        <sub>or this PR leaves requirements*.txt, common.mk and Makefile untouched</sub>
- [x] Added `R` tag to commit title                         <sub>or this PR leaves requirements*.txt untouched</sub>
- [x] Added `reqs` label to PR                              <sub>or this PR leaves requirements*.txt untouched</sub>

Author (before every review)

- [x] `make integration_test` passes in personal deployment <sub>or this PR does not touch functionality that could break the IT</sub>
- [x] Rebased branch on `develop`, squashed old fixups

Primary reviewer (after approval)

- [x] Commented on demo expectations                        <sub>or labelled PR as `no demo`</sub>
- [x] Decided if PR can be labeled `no sandbox`
- [x] PR title is appropriate as title of merge commit
- [x] Moved ticket to Approved column
- [x] Assigned PR to an operator

Operator (before pushing merge the commit)

- [x] Checked `reindex` label and `r` commit title tag
- [x] Checked that Demo expectations are clear              <sub>or PR is labeled as `no demo`</sub>
- [x] Rebased and squashed branch
- [x] Sanity-checked history
- [x] Pushed PR branch to Github
- [x] Branch pushed to Gitlab and build passes in `sandbox` <sub>or added `no sandbox` label</sub>
- [x] Started reindex in `sandbox`                          <sub>or this PR does not require reindexing `sandbox`</sub>
- [x] Checked for failures in `sandbox`                     <sub>or this PR does not require reindexing `sandbox`</sub>
- [x] Added PR reference to merge commit title
- [x] Collected commit title tags in merge commit title
- [x] Moved linked issue to Merged column
- [ ] Pushed merge commit to Github

Operator (after pushing the merge commit)

- [ ] Moved freebies to Merged column                       <sub>or there are no freebies in this PR</sub> 
- [ ] Shortened the PR chain                                <sub>or this PR is not the base of another PR</sub>
- [ ] Verified that `N reviews` labelling is accurate
- [x] Pushed merge commit to Gitlab                         <sub>or this changes can be pushed later, together with another PR</sub>
- [ ] Deleted PR branch from Github and Gitlab

Operator (reindex) 

- [x] Purchased BQ slot committment in `dev`                <sub>or this PR does not require reindexing `dev`</sub>
- [x] Started reindex in `dev`                              <sub>or this PR does not require reindexing `dev`</sub>
- [x] Deleted BQ slot committment in `dev`                  <sub>or this PR does not require reindexing `dev`</sub>
- [x] Checked for failures in `dev`                         <sub>or this PR does not require reindexing `dev`</sub>
- [x] Purchased BQ slot committment in `prod`               <sub>or this PR does not require reindexing `prod`</sub>
- [x] Started reindex in `prod`                             <sub>or this PR does not require reindexing `prod`</sub>
- [x] Deleted BQ slot committment in `prod`                 <sub>or this PR does not require reindexing `prod`</sub>
- [x] Checked for failures in `prod`                        <sub>or this PR does not require reindexing `prod`</sub>

Operator

- [ ] Unassigned PR

### CODE REVIEW

**Review by jessebrennan — APPROVED** [review]
This is an elegant way to handle passing the arguments :+1:

File: scripts/can_bundle.py [hunk 1]
```diff
@@ -44,9 +44,14 @@ def main(argv):
     default_catalog = config.default_catalog
     plugin_cls = RepositoryPlugin.load(default_catalog)
     plugin = plugin_cls.create(default_catalog)
-    default_source = str(first(plugin.sources))
+    try:
+        default_source = str(one(plugin.sources))
+    except ValueError:
```
[the remark this answers is not in the document]
Comment (achave11) [reply scripts/can_bundle.py]: @jessebrennan Agree with your observation and added the suggestion. It doesn't hurt to verify `ValueError`'s provenance from calling `one` on an invalid list.

File: scripts/can_bundle.py [hunk 2]
```diff
@@ -44,9 +44,17 @@ def main(argv):
     default_catalog = config.default_catalog
     plugin_cls = RepositoryPlugin.load(default_catalog)
     plugin = plugin_cls.create(default_catalog)
-    default_source = str(first(plugin.sources))
+    try:
+        default_source = str(one(plugin.sources))
+    except ValueError as e:
+        if 'too many items in iterable (expected 1)' in e.args:
```
Comment (hannes-ucsc) [inline scripts/can_bundle.py:50]: Generally, we don't want to depend on a human-readable error message to make decisions. It's too brittle. You don't *have* to use `one` and EAFP. Sometimes it's easier to examine the input first.

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]

File: scripts/can_bundle.py (hunk 2 above)
Comment (jessebrennan) [reply scripts/can_bundle.py]: Sorry... checking the error message here was [my idea](https://github.com/DataBiosphere/azul/pull/2901#discussion_r595499253).

**Review by hannes-ucsc — APPROVED** [review]

**Review by hannes-ucsc — APPROVED** [review]

**Comment by hannes-ucsc** [comment]
For demo, mock the two cases (one source, more than one source) and show how the script behaves.

**Pull request merged**